packages/react/src/file-upload/utils.test.tsx

```diff
@@ -168,3 +173,29 @@ describe('formatFileExtension', () => {
 		expect(formatFileExtension('.docx')).toEqual('docx');
 	});
 });
+
+describe('sortFileByName', () => {
+	it('sorts files by file name correctly', () => {
+		const A: FileWithPath = createExampleFile({ name: 'A' });
+		const B: FileWithPath = createExampleFile({ name: 'B' });
+		const C: FileWithPath = createExampleFile({ name: 'C' });
+		const files: Array<FileWithPath> = [B, C, A];
+		expect(files.sort(sortFileByName)).toMatchObject([A, B, C]);
+	});
+});
+
+describe('sortRejectionByName', () => {
+	it('sorts rejected files by file name correctly', () => {
+		const createFile = (name: string): RejectedFile => ({
+			file: createExampleFile({ name }),
+			errors: [tooManyFilesError],
+		});
+		const A: RejectedFile = createFile('A');
+		const B: RejectedFile = createFile('B');
+		const C: RejectedFile = createFile('C');
+		const files: Array<RejectedFile> = [B, C, A];
+		expect(files.sort(sortRejectionByName)).toMatchObject([A, B, C]);
+	});
+});
```

Comment:
@stowball The unit tests appear to be running forever on this PR. Can you see a reason why that might be?

Reply:
Hmmm maybe cancel and re-run?

Reply:
On the surface of this, I think your assertion is too expensive. Consider just checking the name property of each file, rather than the entire file.

Reply:
@Dan503 I just tested locally and Jest hangs for me. If we switch to asserting the order of the filenames it completes and I believe will satisfy the requirement of the test. This is how I'd write these two tests (note that the typing can be inferred, which makes things simpler):

```ts
describe('sortFileByName', () => {
	it('sorts files by file name correctly', () => {
		const A = createExampleFile({ name: 'A' });
		const B = createExampleFile({ name: 'B' });
		const C = createExampleFile({ name: 'C' });
		const orderedFileNames = [B, C, A]
			.sort(sortFileByName)
			.map((file) => file.name);
		expect(orderedFileNames).toMatchObject([A.name, B.name, C.name]);
	});
});

describe('sortRejectionByName', () => {
	it('sorts rejected files by file name correctly', () => {
		const createFile = (name: string): RejectedFile => ({
			file: createExampleFile({ name }),
			errors: [tooManyFilesError],
		});
		const A = createFile('A');
		const B = createFile('B');
		const C = createFile('C');
		const sortedFileNames = [B, C, A]
			.sort(sortRejectionByName)
			.map(({ file }) => file.name);
		expect(sortedFileNames).toMatchObject([
			A.file.name,
			B.file.name,
			C.file.name,
		]);
	});
});
```

<img width="962" alt="image" src="https://github.com/agriculturegovau/agds-next/assets/43688949/d5cfc401-afb2-456f-992a-9ef60eaec972">

Reply:
toMatchObject was still taking forever for me even with your optimisation. I ended up using JSON.stringify instead.